### PR TITLE
Remove Candid authentication

### LINKED ISSUE #4: Remove Canonical Candid support
Candid is a dead technology which should be replaced by the OIDC support already in Incus.

### PR DESCRIPTION
**Pull request by monstermunchkin**
Resolves #4

### CODE REVIEW

**Comment by monstermunchkin** [comment]
@stgraber should external authentication via Macaroons also be removed?

**Comment by monstermunchkin** [comment]
@stgraber ready for review. The failed test is because there's no space left on the device.

File: doc/api-extensions.md [hunk 1]
```diff
@@ -420,10 +420,6 @@ files for the container via `nofile`. The format is `limits.kernel.[limit name]`
 
 This adds support for renaming custom storage volumes.
 
-## `external_authentication`
```
Comment (stgraber) [inline doc/api-extensions.md:423]: Interesting that the name of the extension doesn't match shared/version/api.go, maybe send a fix to LXD to line them up so they don't hit an issue at some point?

File: test/godeps.list [hunk 2]
```diff
@@ -1,8 +1,5 @@
-github.com/go-macaroon-bakery/macaroon-bakery/v3/bakery
 github.com/go-macaroon-bakery/macaroon-bakery/v3/httpbakery
```
Comment (stgraber) [inline test/godeps.list:1]: What's still using /httpbakery? I'd have expected it to also go away with this change.

**Comment by stgraber** [comment]
@monstermunchkin I retried the random test and it's all clean now. Branch looks good, just one quick question about what's left of httpbakery and then can merge.

File: test/godeps.list (hunk 2 above)
Comment (monstermunchkin) [reply test/godeps.list:1]: I was using the `OpenWebBrowser` function from httpbakery, but added a similar function to `util.go` now.

File: doc/api-extensions.md (hunk 1 above)
Comment (monstermunchkin) [reply doc/api-extensions.md:423]: I've sent a PR.

**Comment by stgraber** [comment]
test/godeps.list needs updating, that's causing a static analysis failure

**Comment by monstermunchkin** [comment]
@stgraber good to go.

**Review by stgraber — APPROVED** [review]
LGTM, thanks!

**Pull request merged**